EndpointCategory/helpers/utils.js

```diff
@@ -28,56 +28,84 @@ module.exports = {
+   * @param {string} dateType - date type to include in error message for
+   *   specificity
    * @returns {string} formatted date string
    */
-  sanitizeDate: (date) => {
-    // Check if string passed
-    if (typeof date === 'string') {
-      const re = /\d{4}-([1-9]|0[1-9]|1[012])-([1-9]|0[1-9]|[12][0-9]|3[01])$/;
-      // If string is in correct format, return as is
-      if (re.test(date)) {
-        return date;
-      }
-      // Throw error if date doesn't match format
+  formatDate: (date, dateType) => {
+    // Check if date instance passed
+    if (date instanceof Date) {
+      // return formatted date string
+      return date.toISOString()
+        .split('T')[0];
+    }
+
+    // Check if passed parameter can be accepted by the Date constructor
+    const timestamp = Date.parse(date);
+    if (Number.isNaN(timestamp)) {
+      // Param cannot be converted to a Date object so throw an error
       throw new ZACCLError({
-        message: 'Date needs to be in the format "yyyy-mm-dd" or a Date instance',
+        message: `${dateType} needs to be a JS Date instance or a string accepted by the Date constructor`,
         code: ERROR_CODES.INVALID_DATE_FORMAT,
       });
     }
 
-    // Check if Date instance passed
-    if (typeof date.getMonth === 'function') {
-      // Add 1 to month to get 1-12 scale
-      const month = date.getMonth() + 1;
-
-      // Return formatted Date string
-      return `${date.getFullYear()}-${month}-${date.getDate()}`;
-    }
-
-    throw new ZACCLError({
-      message: 'Date needs to be in the format "yyyy-mm-dd" or a Date instance',
-      code: ERROR_CODES.INVALID_DATE_FORMAT,
-    });
+    // Param can be converted to a Date instance so return formatted date
+    return new Date(timestamp).toISOString()
+      .split('T')[0];
   },
 
   /**
-   * Returns parsed int if arg is a valid number,
-   *   throws error otherwise
+   * Returns parsed number if arg is a valid number,
+   *   throws ZACCL Error otherwise
    * @author Aryan Pandey
    * @param {number} num - number that needs to be sanitized
+   * @param {string} param - string to identify param
    * @returns {number} parsed number
    */
-  sanitizeInt: (num) => {
-    const parsedNum = Number.parseInt(num);
-    // If not a number throw a standard error
+  sanitizeNum: (num, param) => {
+    const parsedNum = Number(num);
+    // If not a number throw a ZACCL Error
     if (Number.isNaN(parsedNum)) {
-      throw new Error('Not a number');
+      const code = `INVALID_${param.toUpperCase()}`;
+      throw new ZACCLError({
+        message: `${param} is not a valid number`,
+        code: ERROR_CODES[code],
+      });
     }
     return parsedNum;
   },
+
+  /**
+   * Returns param with corrected type if possible,
+   *   throws ZACCL Error otherwise
+   * @author Aryan Pandey
+   * @param {string} param - string to identify param
+   * @param {any} value - value of param that needs to be converted to
```

Comment:
Can the type of `value` and `type` be switched to `{string|number|boolean|date}` and throw an error if type is not one of these things (after the last `if` just throw an error)? There is a chance that this function will just return an undefined.